### PR TITLE
Make EndorsingBankItemSequenceNumber field from ReturnDetailAddendumD optional

### PR DESCRIPTION
**Pull request by SaraAdineh**

### CODE REVIEW

**Comment by atonks2** [comment]
Thanks for the PR @SaraAdineh!

Looks like I was wrong in Slack about the length of that function name. I didn't realize we were linting test files that way. Would you mind coming up with a shorter name for that?

**Comment by adamdecaf** [comment]
I'm seeing the following error about it being unused, not that the name is too long. 

```
returnDetailAddendumD_test.go:31:6: `mockReturnDetailAddendumDWithoutEndorsingBankItemSequenceNumber` is unused (deadcode)
```

File: returnDetailAddendumD_test.go [hunk 1]
```diff
@@ -73,6 +88,41 @@ func TestMockReturnDetailAddendumD(t *testing.T) {
 	}
 }
 
+// TestMockReturnDetailAddendumD creates a ReturnDetailAddendumD
+func TestMockReturnDetailAddendumDWithoutEndorsingBankItemSequenceNumber(t *testing.T) {
+	rdAddendumD := mockReturnDetailAddendumD()
```
Comment (adamdecaf) [inline returnDetailAddendumD_test.go:93]: Should we be using the mock function here? 

```suggestion
	rdAddendumD := mockReturnDetailAddendumDWithoutEndorsingBankItemSequenceNumber()
```

**Comment by SaraAdineh** [comment]
Sorry for the confusion. That's the culprit of making PR later in the evening. I just added the change. Thank you @adamdecaf and @atonks2 :)

**Comment by adamdecaf** [comment]
```
./lint-project.sh: line 209: bc: command not found
```

That error is with our CI. I'll get a fix pushed.

**Review by atonks2 — APPROVED** [review]

**Pull request merged**